### ISSUE TITLE
Possibility to include bagging_fraction when subsampling?

### ISSUE THREAD

**Issue by palVJ** [dmlc/xgboost#5042]
There exists evidence that for classification problems with imbalanced datasets, subsampling of individuals where the number of samples in each class is the same, improves prediction:
[https://www.sciencedirect.com/science/article/pii/S0167947316301803#s000115](url)
In LightGBM this is possible to do via pos_bagging_fraction/neg_bagging_fraction.
Is there a possibility to somewhere in the future include this as an option/parameter in xgboost as well?

**Comment by hcho3** [comment]
XGBoost already offers option `scale_pos_weight` which lets you assign higher weight to the minority class. The loss function takes account of data weights. What are benefits of sampling with different fraction as supposed to assigning different data weights?

**Comment by palVJ** [comment]
These are my thoughts: As in any case with an imbalanced dataset, it is a problem to find the real differences between any classes if any of the classes are not sufficiently representative no matter what we set scale_pos_weight equal to. For a given tree in XGBoost, we must make sure that the minority classes are as representative as possible. However setting subsample = 1, thereby including all data, is often not benficial with respect to overfitting as well as computational considerations.  It would therefore be an advantage, in addition to the scale_pos_weight parameter,  if one could have the option to choose the proportion of each class to be sampled in each tree.
